src/pool/spawn.rs

```diff
@@ -56,13 +56,29 @@ impl<T> QueueCore<T> {
     /// the action.
     pub fn ensure_workers(&self, source: usize) {
         let cnt = self.active_workers.load(Ordering::SeqCst);
-        if (cnt >> WORKER_COUNT_SHIFT) >= self.config.max_thread_count || is_shutdown(cnt) {
+        if (cnt >> WORKER_COUNT_SHIFT) >= self.config.core_thread_count.load(Ordering::SeqCst)
+            || is_shutdown(cnt)
+        {
             return;
         }
 
         let addr = self as *const QueueCore<T> as usize;
+        let mut unparked_once = false;
+
         unsafe {
-            parking_lot_core::unpark_one(addr, |_| UnparkToken(source));
+            parking_lot_core::unpark_filter(
+                addr,
+                |p: ParkToken| {
+                    if !unparked_once && p.0 <= self.config.core_thread_count.load(Ordering::SeqCst)
+                    {
+                        unparked_once = true;
+                        FilterOp::Unpark
+                    } else {
+                        FilterOp::Skip
```

Comment:
Why not `FilterOp::Stop`?

Reply:
If stop, the next execution will start from the position that caused stop, which will be problematic, for example:

core = 4, max = 16
queue is 1, 11, 2, 3

The first ensure_worker will wakeup 1 and stop at 11, and subsequent ensure_worker will always stop at 11.

Reply:
So, this means the `unpark` is an `O(n)` operation where `n` is the number of parked threads?

Reply:
Do you have benchmark result when `core_thread_count` is much smaller than `max_thread_count` (e.g. 4 out of 32)

Reply:
> Do you have benchmark result when `core_thread_count` is much smaller than `max_thread_count` (e.g. 4 out of 32)

I'll upload it later, thanks!